test/itkSpeckleReducingAnisotropicDiffusionImageFilterTest.cxx

```diff
@@ -0,0 +1,72 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkMirrorPadImageFilter.h"
+#include "itkTestingMacros.h"
+
+#include "itkSpeckleReducingAnisotropicDiffusionImageFilter.h"
+
+int
+itkSpeckleReducingAnisotropicDiffusionImageFilterTest(int argc, char * argv[])
+{
+  if (argc < 3)
+  {
+    std::cerr << "Usage: " << argv[0];
+    std::cerr << " inputImage outputImagePrefix";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  using PixelType = float;
+  const unsigned int Dimension = 2;
+
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  ImageType::Pointer image = itk::ReadImage<ImageType>(argv[1]);
+  
+  using PadFilterType = itk::MirrorPadImageFilter<ImageType, ImageType>;
+  PadFilterType::Pointer padFilter = PadFilterType::New();
+  padFilter->SetInput(image);
+  ImageType::SizeType padSize;
+  padSize[0] = 1;
+  padSize[1] = 1;
+  padFilter->SetPadLowerBound(padSize);
+  padFilter->SetPadUpperBound(padSize);
+  padFilter->Update();
+  
+  using SpeckleFilterType = itk::SpeckleReducingAnisotropicDiffusionImageFilter<ImageType>;
+  SpeckleFilterType::Pointer speckleFilter = SpeckleFilterType::New();
+  speckleFilter->SetInput(padFilter->GetOutput());
+  speckleFilter->SetNumberOfWorkUnits(5);
```

Comment:
Is this weird choice intentional?

Reply:
I chose an arbitrary number of work units to demonstrate multithreading and which gave a reasonable speedup on my local machine. What is a better way to select a number of units?

I will add update comments to clarify image padding requirements.

Reply:
4 or 8 are more logical choices. Also, if you don't manually specify that, it should be chosen by default. If you remove this line, how does that affect execution speed?

Reply:
Makes sense. After removing this line `speckleFilter->GetNumberOfWorkUnits()` returns 48 units and the test has speedup of approximately 6x over single-threaded. I will leave this line out of the test.